rancher/rancher
# [v2.7.1] Remove `rancherd` command and build steps

## Issue: https://github.com/rancher/rancher/issues/34913


## Problem
The `rancherd` command is no longer used within Rancher, but building the command and publishing related artifacts is still apart of the rancher build process.
 
## Solution
Remove the `rancherd` command and build steps within Rancher. This includes DockerFiles, scripts, and drone steps. No source code has been changed or modified in relation to `rancherd` so that existing clusters created via `rancherd` are managed properly. 

## Testing
This PR is basically just removing build steps and build artifacts. The main goal is that the drone build no longer performs any steps for `rancherd` and that no new build artifacts are uploaded to DockerHub. Another goal is that the `rancherd` binary is no longer present in the `bin` directory after running `dapper ci`. 

The `rancherd` build process produced an artifact called `rancher-runtime` which was uploaded to DockerHub, and simply contained the `rancherd` chart. From what I can tell, this image is not used by Rancher or any other project so it should be safe to stop publishing this artifact. [This is where the image would be published](https://hub.docker.com/r/rancher/rancher-runtime/tags). After this change the repository should not offer any new image versions. The artifact was being published on each merge into Rancher, so an RC is not necessary. Additionally, drone should no longer have messages such as `Successfully tagged rancher/rancher-runtime` or other references to `rancher-runtime`. 

To test that the `rancherd` binary is no longer created, run `dapper ci` locally within the Rancher project in your IDE and check the resulting `bin` directory. 

## Engineering Testing
### Manual Testing
I've ran `dapper ci` locally and have confirmed that the output of `bin` is consistent with v2.7-head, with the only difference being the absence of the `rancherd` command and associated charts (`rancher.yaml` & `rancher-namespace.yaml`). I've confirmed that the chart bundle (`.tgz` file) produced by the CI has not changed.

I've ensured that the drone build for this PR is passing and that the relevant steps for publishing `rancherd` are no longer taken when building rancher versions, and that all other steps are passing as expected.

I've looked into other projects which may utilize `rancherd`, such as Harvester, and was unable to find any dependency upon the build steps and build artifacts being removed in this PR. The `rancherd` github repository is still available and releases of the program can be found there. 

### Automated Testing
n/a

## QA Testing Considerations

 
### Regressions Considerations
We should check that all build artifacts and the Rancher chart are properly built and work as expected. I haven't seen any unexpected changes locally, but a smoke test is still desirable.
